### DISCUSSION TITLE
Weird interaction of trouble-v3 and ]q

### DISCUSSION THREAD

**Discussion by dusty-phillips** [LazyVim/LazyVim#3349] [Q&A]
I have the trouble-v3 extra enabled, so I'm not 100% sure if this is a bug report against trouble or against the lazyvim configuration of trouble-v3. Since @folke owns both, I just flipped a coin as to which project to post in.

A flow I like to adopt is:

* use space-/ to search for some text in telescope
* hit ctrl-t to open the search results in trouble
* navigate between search results using ]q and [q

In my current (latest lazyvim with all plugins synced recently) setup, the last bullet point isn't work because trouble automatically moves the "selected" element to the list. So when I hit `]q`, it selects the next element as expected. But it moves it to the top of the list, so when I hit `]q` again, it simply alternates between the top-two elements instead of navigating down the list.

Workaround for now is to not use trouble, so I hit `ctrl-q` from the telescope window to put the items in quickfix instead of trouble.

**Comment by dpetka2001** [comment]
I also confirm this behavior. Not sure if it's a Trouble-v3 bug or some kind of misconfiguration of the LazyVim keymaps compared to stable Trouble.

**Reply by folke** [reply]
Yep, seems you're right. It only seems to happen when navigating quickly.
`mini.animate` might also be causing issues.

Will change the implementation so that I always got to the next item in the list instead of what I do now.

**Reply by dpetka2001** [reply]
I don't have `mini.animate` enabled.

**Reply by dpetka2001** [reply]
I only have the following Extras enabled
```
    ● coding.mini-surround    mini.surround
    ● coding.yanky    yanky.nvim
      Better Yank/Paste
    ● dap.core    mason-nvim-dap.nvim  neoconf.nvim  nvim-dap  nvim-dap-ui  nvim-dap-virtual-text  nvim-nio  which-key.nvim
      Debugging support. Requires language specific adapters to be configured. (see lang extras)
    ● dap.nlua  nvim-dap  one-small-step-for-vimkind
    ● editor.trouble-v3    trouble.nvim  edgy.nvim  lualine.nvim  telescope.nvim
      Trouble rewrite including document symbols and a lualine component
    ● ui.mini-indentscope  indent-blankline.nvim  mini.indentscope
    ● ui.treesitter-context  nvim-treesitter-context
    ● util.dot  mason.nvim  nvim-lspconfig  nvim-treesitter
    ● util.mini-hipatterns    mini.hipatterns
```
Just for reference, in case you think some of these might interfere

**Comment by folke** [comment]
I just pushed some fixes. Let me know if you still have the issue

**Reply by dpetka2001** [reply]
Still the same after update.

**Reply by dpetka2001** [reply]
Are the keymaps set up correctly in LazyVim for v3?

**Reply by folke** [reply]
yep, just double checked.

**Comment by folke** [comment]
https://github.com/LazyVim/LazyVim/issues/3359
